airbyte-integrations/connectors/destination-mysql/src/main/java/io/airbyte/integrations/destination/mysql/MySQLSqlOperations.java

```diff
@@ -0,0 +1,117 @@
+package io.airbyte.integrations.destination.mysql;
+
+import io.airbyte.db.jdbc.JdbcDatabase;
+import io.airbyte.integrations.base.JavaBaseConstants;
+import io.airbyte.integrations.destination.jdbc.DefaultSqlOperations;
+import io.airbyte.protocol.models.AirbyteRecordMessage;
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class MySQLSqlOperations extends DefaultSqlOperations {
+  private boolean isLocalFileEnabled = false;
+
+  @Override
+  public void executeTransaction(JdbcDatabase database, List<String> queries) throws Exception {
+    database.executeWithinTransaction(queries);
+  }
+
+  @Override
+  public void insertRecords(JdbcDatabase database, List<AirbyteRecordMessage> records,
+      String schemaName, String tmpTableName) throws SQLException {
+    if (records.isEmpty()) {
+      return;
+    }
+
+    boolean localFileEnabled = isLocalFileEnabled || checkIfLocalFileIsEnabled(database);
+
+    if(!localFileEnabled) {
+      tryEnableLocalFile(database);
+    }
+    isLocalFileEnabled = true;
+    loadDataIntoTable(database, records, schemaName, tmpTableName);
+  }
+
+  private void loadDataIntoTable(JdbcDatabase database, List<AirbyteRecordMessage> records,
+      String schemaName, String tmpTableName) throws SQLException {
+    database.execute(connection -> {
+      File tmpFile = null;
+      try {
+        tmpFile = Files.createTempFile(tmpTableName + "-", ".tmp").toFile();
+        writeBatchToFile(tmpFile, records);
+
+        String absoluteFile = "'" + tmpFile.getAbsolutePath() + "'";
+
+        String query = String.format(
+            "LOAD DATA LOCAL INFILE %s INTO TABLE %s.%s FIELDS TERMINATED BY ',' ENCLOSED BY '\"' LINES TERMINATED BY '\\r\\n'",
+            absoluteFile, schemaName, tmpTableName);
+
+        try (Statement stmt = connection.createStatement()) {
+          stmt.execute(query);
+        }
+      } catch (Exception e) {
+        throw new RuntimeException(e);
+      } finally {
+        try {
+          if (tmpFile != null) {
+            Files.delete(tmpFile.toPath());
+          }
+        } catch (IOException e) {
+          throw new RuntimeException(e);
+        }
+      }
+    });
+  }
+
+  void tryEnableLocalFile(JdbcDatabase database) throws SQLException {
+    database.execute(connection -> {
+      try (Statement statement = connection.createStatement()) {
+        statement.execute("set global local_infile=true");
+      } catch (Exception e) {
+        throw new RuntimeException("local_infile attribute could not be enabled", e);
```

Comment:
It seems like if people are using an Airbyte-specific user it's unlikely it will have permissions to do this.

We should provide as much context as possible in the `RuntimeException` as possible so it's clear what they need to do.

Something like:
> The DB user X was unable to set the local_infile attribute on the MySQL server. As an admin user, you will need to run "SET GLOBAL local_infile = true" before syncing data with Airbyte.

Reply:
We'll also need to detail this requirement in the docs.